apps/L1E/instagram/instagram-frontend/src/components/providers/AuthProvider.tsx

```
@@ -0,0 +1,77 @@
+"use client";
+import React, { createContext, PropsWithChildren, useContext, useState } from 'react';
+import { gql, useMutation } from '@apollo/client';
+
+export type AuthUser = {
+  id: string;
+  email: string;
+};
+
+type AuthContextValue = {
+  user: AuthUser | null;
+  login: (email: string, password: string) => Promise<void>;
```

ESLint

[error] 'email' is defined but never used. Allowed unused args must match /^_/u.

(no-unused-vars)